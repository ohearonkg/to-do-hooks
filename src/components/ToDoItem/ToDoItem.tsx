import React from "react";

interface ITodoItem {
  /**
   * The id for this to do
   */
  id: string;

  /**
   * Text to be displayed
   */
  text: string;

  /**
   * Function to be called
   * when the user clicks upon
   * this item
   */
  onClickFunction: () => void;
}

const ToDoItem: React.SFC<ITodoItem> = ({ text, onClickFunction, id }) => (
  <div onClick={() => onClickFunction(id)}>{text}</div>
);

export default ToDoItem;
