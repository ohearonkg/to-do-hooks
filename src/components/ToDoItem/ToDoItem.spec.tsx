import "jest-dom/extend-expect";

import { cleanup, fireEvent, render } from "react-testing-library";

import React from "react";
import ToDoItem from "./ToDoItem";

afterEach(cleanup);

/**
 * Variables for testing
 */
const sampleText = "ABC";
const sampleOnClickFunction = () => ({});
const sampleID = "a1b2c3";

const sampleProps = {
  text: sampleText,
  onClickFunction: sampleOnClickFunction,
  id: sampleID
};

describe("To Do Item", () => {
  /**
   * Snapshot
   */
  it("Should match its snapshot", () => {
    const { asFragment } = render(<ToDoItem {...sampleProps} />);
    expect(asFragment()).toMatchSnapshot();
  });

  /**
   * Rendering text
   */
  it("Should render to text passed to its text prop", () => {
    const { getByText } = render(<ToDoItem {...sampleProps} />);
    expect(getByText(sampleText)).toBeInTheDocument();
  });

  /**
   * Calling onClickFunction
   */
  it("Should call its the function passed to its onClickFunction when clicked", () => {
    const sampleFunction = jest.fn();
    const altProps = { ...sampleProps, onClickFunction: sampleFunction };
    const { getByText } = render(<ToDoItem {...altProps} />);
    fireEvent.click(getByText(sampleText));
    expect(sampleFunction).toHaveBeenCalledTimes(1);
  });

  it("Should call its the function passed to its onClickFunction with this items id", () => {
    const sampleFunction = jest.fn();
    const altProps = { ...sampleProps, onClickFunction: sampleFunction };
    const { getByText } = render(<ToDoItem {...altProps} />);
    fireEvent.click(getByText(sampleText));
    expect(sampleFunction).toHaveBeenCalledWith(sampleID);
  });
});
