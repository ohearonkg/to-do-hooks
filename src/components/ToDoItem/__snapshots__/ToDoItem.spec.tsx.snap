// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`To Do Item Should match its snapshot 1`] = `
<DocumentFragment>
  <div>
    ABC
  </div>
</DocumentFragment>
`;
